### PR TITLE
Fix build on alpine arm64

### PR DESCRIPTION
**Pull request by am11**
```sh
  In file included from /runtime/src/coreclr/classlibnative/../vm/cgensys.h:22:
  /runtime/src/coreclr/classlibnative/../vm/arm64/cgencpu.h:307:15: error: assigning to 'TADDR' (aka 'unsigned long') from incompatible type 'std::nullptr_t'
    307 |         mem = NULL;
        |               ^~~~
  /usr/include/time.h:11:14: note: expanded from macro 'NULL'
     11 | #define NULL nullptr
        |              ^~~~~~~
```

Few things blocking non-cross alpine arm64 native build. i.e. `eng/install-native-dependencies.sh && ./build.sh clr+libs+packs`

### CODE REVIEW

File: src/coreclr/vm/arm64/cgencpu.h [hunk 1]
```diff
@@ -304,7 +304,7 @@ inline TADDR GetMem(PCODE address, SIZE_T size, bool signExtend)
     }
     EX_CATCH
     {
-        mem = NULL;
+        mem = nullptr;
```
Comment (am11) [inline src/coreclr/vm/arm64/cgencpu.h:307]:
```suggestion
        mem = 0;
```

File: src/coreclr/pgosupport.cmake [hunk 2]
```diff
@@ -59,9 +56,6 @@ function(add_pgo TargetName)
                         if(HAVE_LTO)
                             target_compile_options(${TargetName} PRIVATE -flto -fprofile-instr-use=${ProfilePath} -Wno-profile-instr-out-of-date -Wno-profile-instr-unprofiled)
                             set_property(TARGET ${TargetName} APPEND_STRING PROPERTY LINK_FLAGS " -flto -fprofile-instr-use=${ProfilePath}")
-                            if(NOT LD_LLVM)
```
Comment (janvorli) [inline src/coreclr/pgosupport.cmake:62]: Why did you remove this? I don't think the default linker supports LTO.

Comment (am11) [reply src/coreclr/pgosupport.cmake:62]: lld doesn't require this condition which most builds are using. bfd, for past ~7+ years supports lto. If there are builds using bfd and requiring pgosupport, those should be covered.

**Review by jkoritzinsky — APPROVED** [review]
My only concerns were with the LTO changes, but it looks like that's been addressed.

LGTM!
